src/libsync/progressdispatcher.cpp

```diff
@@ -201,19 +201,14 @@ void ProgressInfo::updateTotalsForFile(const SyncFileItem &item, quint64 newSize
         return;
     }
 
-    // Update item's progress total
-    setProgressItem(item, 0);
-    _currentItems[item._file]._progress._total = newSize;
-
-    // Update _sizeProgress._total
-    _sizeProgress._total = 0;
-    QHashIterator<QString, ProgressItem> it(_currentItems);
-    while (it.hasNext()) {
-        it.next();
-        if (isSizeDependent(it.value()._item)) {
-            _sizeProgress._total += it.value()._progress._total;
-        }
+    if (!_currentItems.contains(item._file)) {
+        _sizeProgress._total += newSize - item._size;
+        setProgressItem(item, 0);
+    } else {
+        _sizeProgress._total += newSize - _currentItems[item._file]._progress._total;
+        _currentItems[item._file]._progress._total = newSize;
```

Comment:
This needs to move out of the `else` block, otherwise _progress._total is item._size - it should always end up at newSize.

Reply:
indeed.